### PR TITLE
Avro: Schema conversions should preserve field docs

### PR DESCRIPTION
**Pull request by kbendick**
This addresses issue https://github.com/apache/iceberg/issues/1954

### CODE REVIEW

**Comment by kbendick** [comment]
So the issue with the merging manifests test appears to be that the test in question uses a hardcoded value for `commit.manifest.target-size-bytes` based on a size expected _without doc comments_.Because the manifest files have doc comments in them, the overall size of each manifest increases and that affects the merging behavior as outlined in the test. 

It's quite possible that we don't want to make this change (or at least that we don't want the doc comments in the manifest files) due to the increase in size of manifest files. I will let others weigh in on that.

**Comment by kbendick** [comment]
I took a look at the default value for the manifest's target size (8mb) and realized the slight increase in size due to the doc comments in the header of the manifest files aren't really that big of a deal.

I'll clean up the test that was failing (which has the `TODO` comments on it) in the morning and then remove the `WIP` from this PR.

File: core/src/test/java/org/apache/iceberg/TestMergeAppend.java [hunk 1]
```diff
@@ -239,8 +239,10 @@ public void testMergeWithExistingManifest() {
   public void testManifestMergeMinCount() throws IOException {
     Assert.assertEquals("Table should start empty", 0, listManifestFiles().size());
     table.updateProperties().set(TableProperties.MANIFEST_MIN_MERGE_COUNT, "2")
-        // each manifest file is 5227 bytes, so 12000 bytes limit will give us 2 bins with 3 manifest/data files.
-        .set(TableProperties.MANIFEST_TARGET_SIZE_BYTES, "12000")
+        // Each initial v1/v2 ManifestFile is 5661/6397 bytes respectively. Merging two of the given
+        // manifests make one v1/v2 ManifestFile of 5672/6408 bytes respectively, so 12850 bytes
+        // limit will give us two bins with three manifest/data files.
+        .set(TableProperties.MANIFEST_TARGET_SIZE_BYTES, "12850")
```
Comment (kbendick) [inline core/src/test/java/org/apache/iceberg/TestMergeAppend.java:245]: Technically this limit needs to be at least 12817 bytes.

That's one byte larger than the number of bins used in the tests * the size of the largest manifest file in either test run (v1 or v2, v2 is slightly larger) for the set of operations defined in the test to meet their assertions, but I chose a value slightly larger / rounded value to match somewhat how the test was previously set up.

Feel free to suggest a different value.

Comment (rdblue) [reply core/src/test/java/org/apache/iceberg/TestMergeAppend.java:245]: 3x the smaller size would be around 17k, and we need it to be at least about 13k, which is 2x the larger size. I'd probably set this to 15k to split the difference and hopefully avoid needing to update this again as tests change. This is minor, though.

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Thanks, @kbendick! Looks good to me.

File: core/src/test/java/org/apache/iceberg/TestMergeAppend.java (hunk 1 above)
Comment (kbendick) [reply core/src/test/java/org/apache/iceberg/TestMergeAppend.java:245]: I'll open a patch to up this to 15k 👍

Comment (kbendick) [reply core/src/test/java/org/apache/iceberg/TestMergeAppend.java:245]: Opened a PR for the suggested 15k value: https://github.com/apache/iceberg/pull/1999

**Pull request merged**